pkg/providers/oidc/provider_verifier.go

```diff
@@ -116,21 +126,82 @@ type verifierBuilder func(*oidc.Config) *oidc.IDTokenVerifier
 
 func getVerifierBuilder(ctx context.Context, opts ProviderVerifierOptions) (verifierBuilder, DiscoveryProvider, error) {
 	if opts.SkipDiscovery {
+		var keySet oidc.KeySet
+		if opts.JWKsURL != "" {
+			keySet = oidc.NewRemoteKeySet(ctx, opts.JWKsURL)
+		} else {
+			_keySet, err := newKeySetFromStatic(opts.PublicKeys)
+			if err != nil {
+				return nil, nil, fmt.Errorf("error while parsing public keys: %v", err)
+			}
+			keySet = _keySet
+		}
 		// Instead of discovering the JWKs URK, it needs to be specified in the opts already
-		return newVerifierBuilder(ctx, opts.IssuerURL, opts.JWKsURL, opts.SupportedSigningAlgs), nil, nil
+		return newVerifierBuilder(
+			opts.IssuerURL,
+			keySet,
+			opts.SupportedSigningAlgs,
+		), nil, nil
 	}
 
 	provider, err := NewProvider(ctx, opts.IssuerURL, opts.SkipIssuerVerification)
 	if err != nil {
 		return nil, nil, fmt.Errorf("error while discovery OIDC configuration: %v", err)
 	}
-	verifierBuilder := newVerifierBuilder(ctx, opts.IssuerURL, provider.Endpoints().JWKsURL, provider.SupportedSigningAlgs())
+	verifierBuilder := newVerifierBuilder(opts.IssuerURL, oidc.NewRemoteKeySet(ctx, provider.Endpoints().JWKsURL), provider.SupportedSigningAlgs())
 	return verifierBuilder, provider, nil
 }
 
+// ReadFile reads the contents of a file into a byte array.
+func ReadFile(filename string) ([]byte, error) {
```

Comment:
why did you create your own ReadFile wrapper? seems a bit unnecessary? as you only use the method inside the `newKeySetFromStatic`, I would suggest just moving the logic their.